ctam/tests/health_check/basic_health_check_group/ctam_test_redfish_firmware_inventory_collection.py

```diff
@@ -67,16 +70,37 @@ def run(self) -> TestResult:
         """
         actual test verification
         """
-        result = True
         status_msg = ""
+        result = True
+        logger_path = os.path.join(self.dut().logger_path, "RedfishInteropValidator", f"{self.__class__.test_id}_{self.__class__.__name__}")
+
+        #cloning Redfish Interop Validator under temp folder which will be deleted after completion of test case.
         step1 = self.test_run().add_step(f"{self.__class__.__name__} run(), step1")  # type: ignore
         with step1.scope():
-            JSONData = self.group.health_check_ifc.ctam_getfi()
-            if JSONData is None or "error" in JSONData:
-                step1.add_log(LogSeverity.ERROR, f"{self.test_id} : Redfish FW Inventory Collection Check - Failed")
-                result = False
-            else:
-                step1.add_log(LogSeverity.INFO, f"{self.test_id} : Redfish FW Inventory Collection Check - Completed")
+            result = self.git_utils.clone_repo(repo_url="https://github.com/microsoft/Redfish-Interop-Validator.git",
+                                  repo_path="Redfish-Interop-Validator")
+
+        if result:
+            step2 = self.test_run().add_step(f"{self.__class__.__name__} run(), step2")  # type: ignore
+            with step2.scope():
+                # Change this JSON file name and path if you want to run any other excel.
+                source_json = "FirmwareInventory.json"
+                json_file_path = os.path.join(self.dut().default_config_path, source_json)
+                #Running RedfishInteropValidator using json profiles.
+                file_name="RedfishInteropValidator"
+                base_uri = self.dut().uri_builder.format_uri(redfish_str="{GPUMC}",
+                                                                    component_type="GPU")
+                repo_file_name = os.path.join(self.git_utils.repo_path, file_name)
+                
+                result, _ = GitUtils.ctam_redfish_interop_validator(file_name=repo_file_name, connection_url=self.dut().connection_url,
+                                                user_name=self.dut().user_name, user_pass=self.dut().user_pass,
+                                                log_path=logger_path, passthrough=base_uri,
+                                                payload="NodeTree /redfish/v1/UpdateService/FirmwareInventory",
+                                                profile=json_file_path,
+                                                )
+                if not result:
+                    step2.add_log(LogSeverity.ERROR, f"Something went wrong while running redfish command.")
+                step2.add_log(LogSeverity.INFO, f"Redfish Service Command ran successfully and validated.")
```

Comment:
This should be under the else part, otherwise if result is False then also this log will be added.

Reply:
Yes..thats correct. Will get this fixed, Thanks for catching this!!

Reply:
Done. Please check now.